File: dart/algorithms/math/fibonacci.dart
// fibonacci.dart

int fibonacci(int n) {
  if (n <= 1) {
    return n;
  }

  int fib = 1;
  int prevFib = 1;

  for (int i = 2; i < n; i++) {
    int temp = fib;
    fib += prevFib;
    prevFib = temp;
  }

  return fib;
}

void main() {
  int n = 10;
  print("Fibonacci sequence up to $n terms:");
  for (int i = 0; i < n; i++) {
    print(fibonacci(i));
  }
}

File: go/algorithms/math/fibonacci.go
// fibonacci.go

package main

import "fmt"

func fibonacci(n int) []int {
    fibSeq := make([]int, n)

    if n >= 1 {
        fibSeq[0] = 0
    }
    if n >= 2 {
        fibSeq[1] = 1
    }

    for i := 2; i < n; i++ {
        fibSeq[i] = fibSeq[i-1] + fibSeq[i-2]
    }

    return fibSeq
}

func main() {
    n := 10
    fibSeq := fibonacci(n)

    fmt.Printf("Fibonacci sequence of length %d: %v\n", n, fibSeq)
}
